graphql_compiler/query_pagination/__init__.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql.language.printer import print_ast
+
+from graphql_compiler.cost_estimation.cardinality_estimator import estimate_number_of_pages
+from graphql_compiler.query_pagination.parameter_generator import (
+    hydrate_parameters_of_parameterized_query
+)
+from graphql_compiler.query_pagination.query_parameterizer import generate_parameterized_queries
+
+
+PaginationQueries = namedtuple(
+    'PaginationQueries',
+    (
+        'next_page_query',          # Document, AST for the query that along with
+                                    # next_page_parameters is expected to return the next page of
+                                    # page of results of the original query.
+        'next_page_parameters',     # dict, parameters with which to execute the next page query.
+        'continuation_query',       # Document or None, AST for the query that along with
+                                    # continuation_parameters returns the rest of the results of the
+                                    # original query, or None if the original query was not split
+                                    # into two queries.
+        'continuation_parameters',  # dict or None, parameters with which to execute the
+                                    # continuation query, or None if the continuation query was not
+                                    # generated.
+    )
+)
+
+
+def split_into_next_page_query_and_continuation_query(
```

Comment:
As a user of this module, it's not clear to me whether I should use `split_into_next_page_query_and_continuation_query`, or `paginate_query`. Both are public methods in the `__init__.py` module, and their documentations are very similar (I still don't know what's the difference. Is it just the last parameter? Do they do the same thing?)

Reply:
`paginate_query` takes care of finding out how many pages to divide the query into, and to see if the query actually needs to be split. `_split_...` does the meat of the work by adding filters and generating parameters. It doesn't, however, find an estimate for the number of pages letting `paginate_query` do the job.
I've made `split...` into a private function.